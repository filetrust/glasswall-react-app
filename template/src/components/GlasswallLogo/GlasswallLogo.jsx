import React from "react";
import styles from "./GlasswallLogo.module.scss";

const GlasswallLoginLogo = () => {
    return (
        <div className={styles.logo} data-testid="logoDiv"></div>
    );
};

export default GlasswallLoginLogo;