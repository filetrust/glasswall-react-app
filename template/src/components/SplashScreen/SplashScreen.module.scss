.splashContainer {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background: linear-gradient(-45deg, #196480, #0c3450);
}

.logoContainer {
    position: absolute;
    top: 5vh;
    left: 4vw;
    margin: 3rem 1rem;
}

.logo {
    position: absolute;
    top: 5vh;
    left: 4vw;
    width: 20rem;
    height: 10rem;
    margin: 3rem 1rem;

}

.headingContainer {
    width: 50rem;
    max-height: 100vh;
    padding: 1rem;
    overflow-x: hidden;
    overflow-y: auto;
    transition: all .5s;
    text-align: center;
}

.heading {
    position: relative;
    margin-bottom: 0;
    padding-left: 4.2rem;
    font-size: 2.2rem;
    font-weight: normal;
    color: #91e1f3;

    &::after {
        content: '';
        display: block;
        position: absolute;
        top: .5rem;
        left: 0;
        width: 3.3rem;
        height: 1.7rem;
        background-size: 3.3rem 1.7rem;
    }
}

.subHeading {
    @extend .heading;
    color: #ffffff;
    font-size: 1.7rem;
    font-weight: normal;
}

.childContainer {
    @extend .heading;
}