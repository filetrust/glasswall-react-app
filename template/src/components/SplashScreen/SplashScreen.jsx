import React from "react";

import GlasswallLogo from "../GlasswallLogo/GlasswallLogo";

import styles from "./SplashScreen.module.scss";

const SplashScreen = ({ heading, subHeading, children }) => {
    return (
        <div className={styles.splashContainer} data-testid="splashContainerDiv">
            <div className={styles.logoContainer} data-testid="logoContainerDiv">
                <GlasswallLogo />
            </div>
            
            <div className={styles.headingContainer} data-testid="headingContainerDiv">
                <h1 className={styles.heading}>{heading}</h1>
                <h2 className={styles.subHeading}>{subHeading}</h2>
                <div className={styles.childContainer}>
                    {children}
                </div>
            </div>
        </div>
    );
};

export default SplashScreen;
