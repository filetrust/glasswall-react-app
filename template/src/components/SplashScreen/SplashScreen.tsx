import React from "react";

import GlasswallLogo from "../GlasswallLogo/GlasswallLogo";

import styles from "./SplashScreen.module.scss";

export interface SplashScreenProps { heading: string; subHeading?: string; children?: React.ReactNode; }

const SplashScreen = (props: SplashScreenProps) => {
    return (
        <div className={styles.splashContainer} data-testid="splashContainerDiv">
            <div className={styles.logoContainer} data-testid="logoContainerDiv">
                <GlasswallLogo className={styles.logo} />
            </div>

            <div className={styles.headingContainer} data-testid="headingContainerDiv">
                <h1 className={styles.heading}>{props.heading}</h1>
                {props.subHeading &&
                    <h2 className={styles.subHeading}>{props.subHeading}</h2>
                }
                <div className={styles.childContainer} data-testid="childContainerDiv">
                    {props.children}
                </div>
            </div>
        </div>
    );
};

export default SplashScreen;
