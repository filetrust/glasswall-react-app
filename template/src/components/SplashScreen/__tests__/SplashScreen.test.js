import React from "react";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import TestRenderer from "react-test-renderer";
import SplashScreen from "../SplashScreen";
import GlasswallLogo from "../../GlasswallLogo/GlasswallLogo";

//TODO: test correct heading is rendered
//TODO: test correct subheading is rendered
//TODO: test children are rendered correctly

test("SplashScreen_Snapshot", () => {
    // Arrange
    const component = TestRenderer.create(<SplashScreen />);

    // Act
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("Renders_GlasswallLogo_Component", () => {
    const wrapper = shallow(<SplashScreen />);
    expect(wrapper.find(GlasswallLogo)).toHaveLength(1);
});

// Unit Tests
test("Renders_SplashContainer_Div", () => {
    // Arrange
    const { getByTestId } = render(<SplashScreen />);

    // Act
    const logoElement = getByTestId("splashContainerDiv");

    // Assert
    expect(logoElement).toBeInTheDocument();
});

test("Renders_LogoContainer_Div", () => {
    // Arrange
    const { getByTestId } = render(<SplashScreen />);

    // Act
    const logoElement = getByTestId("logoContainerDiv");

    // Assert
    expect(logoElement).toBeInTheDocument();
});

test("Renders_HeadingContainer_Div", () => {
    // Arrange
    const { getByTestId } = render(<SplashScreen />);

    // Act
    const logoElement = getByTestId("headingContainerDiv");

    // Assert
    expect(logoElement).toBeInTheDocument();
});