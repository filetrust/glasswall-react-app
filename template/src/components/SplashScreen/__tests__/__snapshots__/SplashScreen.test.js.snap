// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen_Snapshot 1`] = `
<div
  className="splashContainer"
  data-testid="splashContainerDiv"
>
  <div
    className="logoContainer"
    data-testid="logoContainerDiv"
  >
    <div
      className="logo"
      data-testid="logoDiv"
    />
  </div>
  <div
    className="headingContainer"
    data-testid="headingContainerDiv"
  >
    <h1
      className="heading"
    />
    <h2
      className="subHeading"
    />
    <div
      className="childContainer"
    />
  </div>
</div>
`;
